tests/test_validate_distill.py

```
@@ -0,0 +1,366 @@
+                self.assertEqual(oc.check_digest(record, "x"), [])
+
+    def test_findings_are_stamped_as_failed(self):
+        with tempfile.TemporaryDirectory() as tmp:
+            root = Path(tmp) / "run"
+            record = fr.build_records(3, 1)[0]
+            record["result"]["reason_codes"] = []
+            record["provenance"]["record_sha256"] = oc.record_digest(record)
+            oc.write_jsonl(root / "a.jsonl", [record])
+            out = Path(tmp) / "stamped.jsonl"
+            with contextlib.redirect_stdout(io.StringIO()):
+                with contextlib.redirect_stderr(io.StringIO()):
```

Ruff (0.16.3)

[warning] Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)